lms/models/json_settings.py

```diff
@@ -3,24 +3,52 @@
 from typing import Any, Optional, Tuple
 
 import sqlalchemy as sa
+from pyramid.settings import asbool
 from sqlalchemy.ext.mutable import MutableDict
 from sqlalchemy.orm import InstrumentedAttribute
 
 
+
+class _JSONSettingType:
+    def format_for_storage(self, value):
+        raise NotImplementedError("You cannot store this type directly")
+
+
+class _StringOrNone(_JSONSettingType):
+    def format_for_storage(self, value):
+        if value:
+            value = value.strip()
+
+        if not value:
+            return None
+
+        return value
```

Comment:
Doing this instead of `str` is actually good because it means you can call it without doing anything fancy first. So the two cases become AES or not.